### PR TITLE
OP-1964: Adding network_configs to change download location source for protocol versions.

### PR DESCRIPTION
**Pull request by sacherjj**
Using network_configs folder in debian for allowing multiple network names and source URLs.  This will allow others to submit PRs to allow using the launcher package to pull configs and binaries from other networks.

Also updated config_from_example.sh to allow a second argument to specify IP for NAT installs or when external auto-detection fails.

Added check that these are run as `casper` user to eliminate permissions issues when this is not done. 

Small fix from clippy update.

### CODE REVIEW

File: resources/maintainer_scripts/network_configs/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+mkdir config
+curl -JLO genesis.casperlabs.io/casper/1_0_0/bin.tar.gz
+curl -JLO genesis.casperlabs.io/casper/1_0_0/config.tar.gz
+mv config.tar.gz config_old.tar.gz
+cd config
+tar -xzvf ../config_old.tar.gz
+```
+
+You would need to customize `chainspec.toml` with a new network name and activation_point timestamp.
+You would need to customize `config-example.toml" with new known_addresses.
```
Comment (sacherjj) [inline resources/maintainer_scripts/network_configs/README.md:56]:
```suggestion
You would need to customize `config-example.toml` with new known_addresses.
```

File: resources/maintainer_scripts/network_configs/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+`config.tar.gz` is decompressed into `/etc/casper/<protocol_version>`.
+`bin.tar.gz` is decompressed into `/var/lib/casper/<protocol_version>`.
+
+The script will error if protocol versions already exist.
+
+## Packaging
+
+With merges to `master`, `release-*` and `dev` branches in the `casper-node` repo, the artifacts are created in
+`genesis.casperlabs.io/drone/<git_hash>/<protocol varsion>/[bin.tar.gz|config.tar.gz]`.
```
Comment (Fraser999) [inline resources/maintainer_scripts/network_configs/README.md:40]:
```suggestion
`genesis.casperlabs.io/drone/<git_hash>/<protocol version>/[bin.tar.gz|config.tar.gz]`.
```

**Review by Fraser999 — APPROVED** [review]
Only a single nitpick from me.  I didn't actually run the bash scripts locally, but it looks ok to me (note: I'm no bash expert though!)

**Comment by sacherjj** [comment]
bors merge

**Pull request merged**